src/contracting/storage/orm.py

```diff
@@ -138,4 +148,127 @@ def __getitem__(self, item):
         return super().__getitem__(item)
 
     def clear(self, *args):
-        raise Exception('Cannot write with a ForeignHash.')
+        raise Exception("Cannot write with a ForeignHash.")
+
+
+class LogEvent:
+    """
+    TODO
+    - Break validation into smaller functions
+    - Add checks for use of illegal types and argument names (See Hash checks.)
+    """
+
+    def __init__(self, contract, name, event, params, driver: Driver = driver):
+
+        self._driver = driver
+        self._params = params
+        self._contract = contract
+        self._event = event
+
+        assert isinstance(params, dict), "Args must be a dictionary."
+        assert len(params) > 0, "Args must have at least one argument."
+        # Check for indexed arguments with a maximum of three
+        indexed_args_count = sum(1 for arg in params.values() if arg.get("idx", False))
+        assert (
+            indexed_args_count <= 3
```

Comment:
maximum 3 indexed arguments. the intention is to save these in a JSONB field w/ GIN indexing on the BDS